### PR TITLE
Adding performer fine-tuning research exampke

### PR DESCRIPTION
**Pull request by TevenLeScao**
# What does this PR do?

This PR adds a performer fine-tuning research example based on `run_mlm_flax.py`. The user can fine-tune a Performer/FAVOR+ Bert starting from the Bert checkpoint or blank model of their choice, and compare it to a vanilla Bert model, also from a checkpoint or blank.

@patrickvonplaten

### CODE REVIEW

**Review by patrickvonplaten — APPROVED** [review]
Great! Think it's a good idea to first add the `Performer` model under `research projects` and then at a later stage add it to the main lib.

File: examples/research_projects/performer/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Performer fine-tuning
+
+Example authors: @TevenLeScao, @Patrickvonplaten
+
+Paper authors: Krzysztof Choromanski, Valerii Likhosherstov, David Dohan, Xingyou Song, Andreea Gane, Tamas Sarlos, Peter Hawkins, Jared Davis, Afroz Mohiuddin, Lukasz Kaiser, David Belanger, Lucy Colwell, Adrian Weller
```
Comment (patrickvonplaten) [inline examples/research_projects/performer/README.md:5]: maybe add paper link here somewhere as well: https://arxiv.org/abs/2009.14794

**Comment by patrickvonplaten** [comment]
@sgugger - we want to do some fine-tuning experiments with the new performer model: https://arxiv.org/abs/2009.14794 before adding it to the `src/transformers/`. I think this is a good first place for it where we don't have to be super careful about the API choices yet. Is that fine for you?

**Comment by sgugger** [comment]
Yes, this completely works for me!

**Pull request merged**